x-pack/filebeat/docs/inputs/input-httpjson.asciidoc

```diff
@@ -35,92 +43,322 @@ Example configurations:
+
+Some built-in helper functions are provided to work with the input state inside value templates:
+
+- `parseDuration`: parses duration strings and returns `time.Duration`. Example: `{{parseDuration "1h"}}`.
+- `now`: returns the current `time.Time` object in UTC. Optionally, it can receive a `time.Duration` as a parameter. Example: `{{now (parseDuration "-1h")}}` returns the time at 1 hour before now.
+- `parseTimestamp`: parses a timestamp in seconds and returns a `time.Time` in UTC. Example: `{{parseTimestamp 1604582732}}` returns `2020-11-05 13:25:32 +0000 UTC`.
+- `parseTimestampMilli`: parses a timestamp in milliseconds and returns a `time.Time` in UTC. Example: `{{parseTimestamp 1604582732000}}` returns `2020-11-05 13:25:32 +0000 UTC`.
+- `parseTimestampNano`: parses a timestamp in nanoseconds and returns a `time.Time` in UTC. Example: `{{parseTimestamp 1604582732000000000}}` returns `2020-11-05 13:25:32 +0000 UTC`.
+- `parseDate`: parses a date string and returns a `time.Time` in UTC. By default the expected layout is `RFC3339` but optionally can accept any of the Golangpredefined layouts or a custom one. Example: `{{ parseDate "2020-11-05T12:25:32Z" }}`, `{{ parseDate "2020-11-05T12:25:32.1234567Z" "RFC3339Nano" }}`, `{{ (parseDate "Thu Nov  5 12:25:32 +0000 2020" "Mon Jan _2 15:04:05 -0700 2006").UTC }}`.
+- `formatDate`: formats a `time.Time`. By default the format layout is `RFC3339` but optionally can accept any of the Golang predefined layouts or a custom one. It will default to UTC timezone when formatting, but you can specify a different timezone. If the timezone is incorrect, it will default to UTC. Example: `{{ formatDate (now) "UnixDate" }}`, `{{ formatDate (now) "UnixDate" "America/New_York" }}`.
+- `getRFC5988Link`: extracts a specific relation from a list of https://tools.ietf.org/html/rfc5988[RFC5988] links. It is useful when parsing header values for pagination. Example: `{{ getRFC5988Link "next" .last_response.header.Link }}`.
+- `toInt`: converts a string to an integer. Returns 0 if the conversion fails.
+- `add`: adds a list of integers and returns their sum.
+
+In addition to the provided functions, any of the native functions for `time.Time` and `http.Header` types can be used on the corresponding objects. Examples: `{{(now).Day}}`, `{{.last_response.header.Get "key"}}`
 
 ==== Configuration options
 
 The `httpjson` input supports the following configuration options plus the
 <<{beatname_lc}-input-{type}-common-options>> described later.
 
 [float]
-==== `api_key`
+==== `config_version`
 
-API key to access the HTTP API. When set, this adds an `Authorization` header to
-the HTTP request with this as the value.
+Defines the configuration version. Current supported versions are: `1` and `2`. Default: `1`.
+
+NOTE:  This setting defaults to `1`  to avoid breaking current configurations. V1 configuration is deprecated and will be unsupported in future releases. Any new configuration should use `config_version: 2`.
```

Comment:
```suggestion
NOTE: This setting defaults to `1` to avoid breaking current configurations. V1 configuration is deprecated and will be unsupported in future releases. Any new configuration should use `config_version: 2`.
```